atest/testdata/variables/environment_variables.robot

```diff
@@ -86,6 +86,14 @@ Empty Environment Variable
     [Documentation]    FAIL    STARTS: Environment variable '\%{}' not found.
     Log  %{}
 
+Environment Variable with Default Value
+    ${value}        Set variable    %{NON_EXISTING_VAR=default value}
+                    Should be equal    ${value}    default value
+
+Environment Variable with Empty Default Value
+    ${value}        Set variable    %{NON_EXISTING_VAR=}
+                    Should be equal    ${value}    ${EMPTY}
+
```

Comment:
- You need to add matching tests also to the `atest/robot` side. See CONTRIBUTING.rst and atest/README.rst for details.
- Test with an internal existing variable like `%{NON_EXISTING_VAR=${VAR}}` would be good.
- Should also add a test for a Java properties that can be accessed using this same syntax.
- Testing `=` in the default value would be good as well.
- Please use same style with assignment and other test code as elsewhere.

Reply:
Unfortunately, writing docs is not my best sidde, but i tried to.